### DISCUSSION TITLE
How to put it to loading state externally?

### DISCUSSION THREAD

**Discussion by sarim** [orchidjs/tom-select#216] [Q&A] [answered]
version: 2.0.0-rc.4
I'm doing this.
```js
const tom = new TomSelect(this.element, {
    valueField: 'id',
    labelField: 'name',
    searchField: 'name',
    maxItems: 1
});

try {
    // TODO: tom.setLoadingState();
    const response = await fetch(this.urlValue);
    const json = await response.json();
    tom.setupOptions(json);
    // TODO: tom.setFinishedLoadingState();
} catch (error) {
}

```

Basically I need to populate tomselect externally, and those are remote/asynchronous, so they are waiting I want to put tomselect to the loading state.

I tried `tom.loading = 1;` , It does put it in a loading state AKA the spinner shows up. But then I can't get it out of loading state, calling `tom.loading = 0;` after `tom.setupOptions(json)` doesn't work, the spinner keeps spinning. If I unfocus and then focus again then the options show up.

Basically I'm looking for a less hacky way to control the loading state.

**Comment by oyejorge** [comment, marked as answer]
Take a look at the "load once" example on https://tom-select.js.org/examples/remote/

**Reply by sarim** [reply]
Thanks. Unsetting load (`settings.load = null`) was the missing bit. I managed to implement what I wanted. :) Its basically a chain of dependent fields.
